### PR TITLE
gdal raster zonal-stats: avoid integer overflows on geometries with huge coordinate values

### PR DESCRIPTION
**Pull request by rouault**
CC @dbaston  #13376 insipired me to test with even more edge cases...

### CODE REVIEW

File: alg/zonal.cpp [hunk 1]
```diff
@@ -1541,22 +1478,24 @@ class GDALZonalStatsImpl
...
-                            &oGeomExtent);
+                        poGeom->getEnvelope(&oGeomExtent);
+                        oGeomExtent.Intersect(oChunkExtent);
                         if (!m_srcInvGT.Apply(oGeomExtent, oGeomWindow))
                         {
                             return false;
                         }
-                        TrimWindow(oGeomWindow, oChunkWindow);
-                        OGREnvelope oTrimmedEnvelope = ToEnvelope(oGeomWindow);
+                        if (oGeomWindow.nXSize == 0 || oGeomWindow.nYSize == 0)
```
Comment (dbaston) [inline alg/zonal.cpp]: I was doing `TrimWindow` in pixel/line space because I worried about floating-point issues producing an `oGeomWindow` that is outside the extent of the raster by 1 pixel. (`nXOff == -1` or `nXSize` == `RasterXSize + 1`). Do you think that is a concern?

Comment (rouault) [reply alg/zonal.cpp]: > was doing `TrimWindow` in pixel/line space because I worried about floating-point issues producing an `oGeomWindow` that is outside the extent of the raster by 1 pixel. (`nXOff == -1` or `nXSize` == `RasterXSize + 1`). Do you think that is a concern?

I initially got test failures, that I fixed by tweaking the ceil() and floor() to include an espilon : https://github.com/OSGeo/gdal/pull/13380/files#diff-8ec18576e00ae96f5d026b3cb557e526eb2f6553cb84e9323b131d3c8df2a876R5864

And I've just added the following post checks after m_srcInvGT.Apply() to be sure we fit in the chunk window:
```
                        oGeomWindow.nXOff =
                            std::max(oGeomWindow.nXOff, oChunkWindow.nXOff);
                        oGeomWindow.nYOff =
                            std::max(oGeomWindow.nYOff, oChunkWindow.nYOff);
                        oGeomWindow.nXSize =
                            std::min(oGeomWindow.nXSize,
                                     oChunkWindow.nXOff + oChunkWindow.nXSize -
                                         oGeomWindow.nXOff);
                        oGeomWindow.nYSize =
                            std::min(oGeomWindow.nYSize,
                                     oChunkWindow.nYOff + oChunkWindow.nYSize -
                                         oGeomWindow.nYOff);
```

Comment (dbaston) [reply alg/zonal.cpp]: The same post-check would be needed around line 1660, no?

Comment (rouault) [reply alg/zonal.cpp]: > The same post-check would be needed around line 1660, no?

good point. added in dbaa44d1f7e447195461b77fa4baf455edb938e1
